packages/contracts/test/helpers/multisig.ts

```diff
@@ -1,61 +1,103 @@
+import {
+  deployContract,
+  highGasLimit,
+  signMessageBytes
+} from "@counterfactual/test-utils";
 import * as ethers from "ethers";
 
-export const Operation = Object.freeze({
-  Call: 0,
-  Create: 2,
-  Delegatecall: 1
-});
-
-/**
- * Executes a transaction from the multisig to itself according to
- * to the given data.
- *
- * @param data is the contract abi to invoke the correct function with args.
- * @param multisig is the multisig sending and receiving the tx.
- * @param wallets is the array of wallets signing off on the tx.
- * @param op is the Operation to use, i.e., Call, Delegatecall, or Create.
- */
-export async function executeTxData(data, toAddr, multisig, wallets, op) {
-  const value = 0;
-
-  const transactionHash = await multisig.getTransactionHash(
-    toAddr,
-    value,
-    data,
-    op
-  );
-
-  const signatures = sign(transactionHash, wallets);
-  return multisig.execTransaction(
-    toAddr,
-    value,
-    data,
-    op,
-    signatures.v,
-    signatures.r,
-    signatures.s,
-    {
-      gasLimit: 4712388
-    }
-  );
+const enum Operation {
+  Call = 0,
+  Delegatecall = 1
 }
 
-export function sign(data, wallets) {
-  const sortedWallets = wallets.slice().sort((w1, w2) => {
-    return w1.address < w2.address ? -1 : w1.address === w2.address ? 0 : 1;
-  });
-  const v: number[] = [];
-  const r: string[] = [];
-  const s: string[] = [];
-  sortedWallets.forEach(wallet => {
-    const sig = new ethers.SigningKey(wallet.privateKey).signDigest(data);
-    if (!sig.recoveryParam) {
-      throw new Error("could not sign with wallet");
+export default class Multisig {
+  private static loadTruffleContract() {
+    const MinimumViableMultisig = artifacts.require("MinimumViableMultisig");
+    const Signatures = artifacts.require("Signatures");
+    MinimumViableMultisig.link("Signatures", Signatures.address);
+    return MinimumViableMultisig;
+  }
+
+  private contract?: ethers.Contract;
+
+  constructor(readonly owners: string[]) {}
+
+  get address() {
+    if (!this.contract) {
+      throw new Error("Must deploy Multisig contract first");
+    }
+    return this.contract.address;
+  }
+
+  public async deploy(
+    providerOrSigner: ethers.ethers.Wallet | ethers.types.MinimalProvider
+  ) {
+    const contract = await deployContract(
```

Comment:
Should we do this through the ProxyFactory instead?

Reply:
Not necessary for this test IMO